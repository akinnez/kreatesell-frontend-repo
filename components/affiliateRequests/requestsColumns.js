import Tags from 'components/Tags';
import Performance from 'components/affiliates/Performance';
import {dateString} from 'utils/dateFormat';
import formatNumber from 'utils/formatNumber';
import productPriceFn from 'utils/productPriceFn';
import GetLink from './components/GetLink';
import RequestStatus from './components/RequestStatus';

const requestsColumns = [
	{
		title: 'Product',
		dataIndex: 'product_name',
	},
	{
		title: 'Product Type',
		dataIndex: 'product_type_name',
	},
	{
		title: 'Launch Date',
		dataIndex: 'launch_date',
		render: (dateStr) => dateString(dateStr),
	},
	{
		title: 'Sales Price',
		render: (record) => {
			const priceDetails = productPriceFn(
				record.kreator_product_price_details
			);
			return !priceDetails
				? 0
				: `${priceDetails.currency} ${formatNumber(
						priceDetails.price
				  )}`;
		},
	},
	{
		title: 'Performance',
		render: (record) => (
			<Performance
				sold={record.total_sold}
				visits={record.total_product_visits}
			/>
		),
	},
	{
		title: 'Commission (%)',
		dataIndex: 'affiliate_percentage_on_sales',
		render: (commission) => `${Math.abs(commission)}%`,
	},
	{
		title: 'Request Status',
		dataIndex: 'request_status',
		render: (status) => <RequestStatus status={status} />,
	},
	{
		title: 'Actions',
		render: (record) => (
			<GetLink
				status={record.request_status}
				productId={record.product_id}
			/>
		),
		width: '97px',
	},
];

export default requestsColumns;
