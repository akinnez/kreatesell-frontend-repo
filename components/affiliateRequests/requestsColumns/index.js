import Link from "next/link";
import { MdOutlineLink } from "react-icons/md";
import Tags from "components/Tags";
import Performance from "components/affiliates/Performance";
import { dateString } from "utils/dateFormat";
import formatNumber from "utils/formatNumber";
import styles from "./index.module.scss";

const requestsColumns = [
  {
    title: "Product",
    dataIndex: "product_name",
  },
  {
    title: "Product Type",
    dataIndex: "product_type_name",
  },
  {
    title: "Launch Date",
    dataIndex: "launch_date",
    render: dateStr => (!dateStr ? "N/A" : dateString(dateStr)),
    sorter: (a, b) => new Date(a.date_created) - new Date(b.date_created),
    sortDirections: ["descend", "ascend", "descend"],
  },
  {
    title: "Sales Price",
    render: record => {
      return `${record.currency || ""} ${formatNumber(
        record.sales_price || 0
      )}`;
    },
  },
  {
    title: "Performance",
    render: record => (
      <Performance
        sold={record.total_sold}
        visits={record.total_product_visits}
      />
    ),
  },
  {
    title: "Commission (%)",
    dataIndex: "affiliate_percentage_on_sales",
    render: commission => `${commission}%`,
  },
  {
    title: "Request Status",
    dataIndex: "request_status",
    render: status => (
      <>
        {status === "Approved" ? (
          <Tags color="green">{status}</Tags>
        ) : status === "Declined" ? (
          <Tags color="red">{status}</Tags>
        ) : status === "Pending" ? (
          <Tags color="orange">{status}</Tags>
        ) : (
          <Tags>{status}</Tags>
        )}
      </>
    ),
  },
  {
    title: "Actions",
    render: record => (
      <>
        {record.request_status === "Approved" ? (
          <Link href={`/account/affiliate/requests/${record.product_id}`}>
            <a className={styles.link}>
              Get Link&nbsp; <MdOutlineLink />
            </a>
          </Link>
        ) : (
          <a className={styles.link} disabled>
            Get Link&nbsp; <MdOutlineLink />
          </a>
        )}
      </>
    ),
    width: "97px",
  },
];

export default requestsColumns;
