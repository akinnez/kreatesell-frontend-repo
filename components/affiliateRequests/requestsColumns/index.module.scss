@use "public/css/colors";

.link {
  font-size: 12px;
  font-weight: bold;
  color: colors.$green-100;
  width: 104px;
  display: inline-flex;
  align-items: center;
  padding: 5px 15px;
  border-radius: 8px;
  border: 1px solid colors.$green-100;

  &[disabled] {
    color: #bfbfbf;
    border-color: #bfbfbf;
  }

  svg {
    display: inline;
    font-size: 20px;
  }
}
