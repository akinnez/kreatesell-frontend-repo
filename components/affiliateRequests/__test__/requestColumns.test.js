import requestsColumns from '../requestsColumns';

it('renders a snapshot of the requestsColumns', () => {
	expect(requestsColumns()).toMatchSnapshot();
});
