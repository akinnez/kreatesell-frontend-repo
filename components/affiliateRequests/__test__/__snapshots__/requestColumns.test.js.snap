// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of the requestsColumns 1`] = `
Array [
  Object {
    "dataIndex": "product_name",
    "title": "Product",
  },
  Object {
    "dataIndex": "product_type",
    "render": [Function],
    "title": "Product Type",
  },
  Object {
    "dataIndex": "date_created",
    "render": [Function],
    "sortDirections": Array [
      "descend",
      "ascend",
      "descend",
    ],
    "sorter": [Function],
    "title": "Launch Date",
  },
  Object {
    "dataIndex": "sales_price",
    "render": [Function],
    "title": "Sales Price",
  },
  Object {
    "dataIndex": "performance",
    "render": [Function],
    "title": "Performance",
  },
  Object {
    "dataIndex": "commission",
    "render": [Function],
    "title": "Commission (%)",
  },
  Object {
    "dataIndex": "status",
    "render": [Function],
    "title": "Request Status",
  },
  Object {
    "render": [Function],
    "title": "Actions",
  },
]
`;
